### ISSUE TITLE
Tests of tolerance == 0 and other edge cases is needed

### ISSUE THREAD

**Issue by mmehr2** [mmehr2/adjuster#5]
Those would cause the old code to go into infinite loops.
1. tolerance <= 0.0
2. target < min.luminance OR target > max.luminance

We need regression tests in case those conditions get re-introduced.

**Comment by mmehr2** [comment]
I forgot to include the "fixes #5" comment in the commit for this. (One done since, at this point.)
I did add the Fix#5 tag on that commit, but that is not automatically noted here.
I also will add a comment here about issues encountered trying to make static values for unit testing.

**Comment by mmehr2** [comment]
I tried to add a NUNIT test that would test a range of values, using the constants that were obtained from calls to GetMin/MaxLuminance() on the ISensorControl interface. Unfortunately, this led to the following conclusions:

1. NUNIT [Range(lo, hi, step)] can only be used with constants (oh for C++ constexpr here!)
1. Was there a way to get constant values from a C# Interface? Yes, introduced in C#8 (now on 9), but this does not satisfy the requirements for a constant expression I don't believe (hope I didn't misunderstand this)
1. If I did find a way to do that, we could have minLum and maxLum private statics in ISensorControl, with a SetLumRange(lo, hi) static method (body provided) to set it, and properties (as get; private set;) to access them. Then each implementing class must call the SetLumRange() to set things up if an appropriate way can be found. This seems to be a design smell, though.

Anyway, at my current level of understanding of C# (first week of use), I don't think I am confident enough to pull this off.
